src/python/pants/util/meta.py

```diff
@@ -125,6 +126,15 @@ def freeze_instance(self) -> None:
     def unfreeze_instance(self) -> None:
         self._is_frozen = False
 
+    @contextmanager
+    def unfrozen(self) -> Iterator:
```

Comment:
Maybe we should name this something like `unsafe_unfrozen()` or `unfrozen_for_tests()`? I don't think we want this used in production. While pantsbuild/pants contributors will know better, plugin authors might not.

Reply:
It's added as `_unfrozen`, and you need to do `cast(Any, frozen_instance).unfrozen()`, so I don't think there's a good chance that people will find this unless they go looking.